src/models/userModel.ts

```
@@ -9,6 +9,7 @@ export interface IUserTemplate {
 	passwordConfirm: string | undefined;
 	role: string;
 	_id?: any;
```

I think it will be a string?

---

Have to be any because 
```  _id?: any;```(in mongoose d.ts)
So when 
```IUserTemplate { _id?: string} ```
```Interface 'IUser' cannot simultaneously extend types 'Document' and 'IUserTemplate'.``` Because that difference.
IUser cannot override  that type (just to avoid error) because _id field must be optional. When document based on that interface is created you don't have that the id yet. 

Solution would be to repeat _id?: string; in IUser interface (just to suppress error in userModel) and then fix all errors in other places when user._id would be in inferred as string | undefined